hack/legacy_fill_db/filldb.go

```diff
@@ -98,12 +99,15 @@ func listFolders(partition, numPartitions int) ([]string, error) {
 	}
 
 	lines := strings.Split(string(output), "\n")
-	var folders []string
-	for i, line := range lines {
-		if i%(numPartitions) == partition {
-			folders = append(folders, strings.TrimSpace(line))
+	folders := slices.Collect(func(yield func(string) bool) {
+		for i, line := range lines {
+			if i%(numPartitions) == partition {
+				if !yield(strings.TrimSpace(line)) {
+					return
+				}
+			}
```

Comment:
Using iterators is worse. The code is more complicated than the original code and I'm not sure what it does. Lets keep the simple can clear code.